src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -391,6 +393,41 @@ static Solution UpdateExistingDocumentsToChangedDocumentContents(Solution soluti
         }
     }
 
+    private Solution InitializeAnalyzerFallbackOptions(Solution oldSolution, Solution newSolution)
+    {
+        var newFallbackOptions = newSolution.FallbackAnalyzerOptions;
+
+        // Clear out languages that are no longer present in the solution.
+        // If we didn't, the workspace might clear the solution (which removes the fallback options)
+        // and we would never re-initialize them from global options.
+        foreach (var (language, _) in oldSolution.SolutionState.ProjectCountByLanguage)
```

Comment:
primarily i mainly care if thse interesting cases are tested.

Reply:
Yes, we do have test covering these cases.

Reply:
perfect.